config/src/iroha.rs

```diff
@@ -0,0 +1,154 @@
+        #[config(inner)]
+        pub logger: logger::Configuration,
+        /// `GenesisBlock` configuration
+        #[config(inner)]
+        #[view(into = genesis::ConfigurationView)]
+        pub genesis: genesis::Configuration,
+        /// `WorldStateView` configuration
+        #[config(inner)]
+        pub wsv: wsv::Configuration,
+        /// Network configuration
+        #[config(inner)]
+        pub network: network::Configuration,
+        /// Telemetry configuration
+        #[config(inner)]
+        pub telemetry: telemetry::Configuration,
+    }
+}
+
+impl Default for Configuration {
+    fn default() -> Self {
+        let sumeragi_configuration = sumeragi::Configuration::default();
+        let (public_key, private_key) = sumeragi_configuration.key_pair.clone().into();
+
+        Self {
+            public_key,
+            private_key,
+            disable_panic_terminal_colors: bool::default(),
+            shutdown_on_panic: false,
+            kura: kura::Configuration::default(),
+            sumeragi: sumeragi_configuration,
+            torii: torii::Configuration::default(),
+            block_sync: block_sync::Configuration::default(),
+            queue: queue::Configuration::default(),
+            logger: logger::Configuration::default(),
+            genesis: genesis::Configuration::default(),
+            wsv: wsv::Configuration::default(),
+            network: network::Configuration::default(),
+            telemetry: telemetry::Configuration::default(),
+        }
+    }
+}
+
+impl Configuration {
+    /// Construct [`Self`] from a path-like object.
+    ///
+    /// # Errors
+    /// - File not found.
+    /// - File found, but peer configuration parsing failed.
+    /// - Length of the array in raw json representation is different
+    /// to the lenght of the array in
+    /// [`self.sumeragi.trusted_peers.peers`], most likely due to two
+    /// (or more) peers having the same public key.
+    pub fn from_path<P: AsRef<Path> + Debug + Clone>(path: P) -> Result<Configuration> {
+        let file = File::open(path.clone())
+            .wrap_err(format!("Failed to open the config file {:?}", path))?;
+        let reader = BufReader::new(file);
+        let mut configuration: Configuration = serde_json::from_reader(reader).wrap_err(
+            format!("Failed to parse {:?} as Iroha peer configuration.", path),
+        )?;
+        configuration.finalize()?;
+        Ok(configuration)
+    }
+
+    fn finalize(&mut self) -> Result<()> {
+        self.sumeragi.key_pair = KeyPair::new(self.public_key.clone(), self.private_key.clone())?;
+        self.sumeragi.peer_id =
+            iroha_data_model::peer::Id::new(&self.torii.p2p_addr, &self.public_key.clone());
+
+        Ok(())
+    }
+
+    /// Loads configuration from environment
+    ///
+    /// # Errors
+    /// If Configuration deserialize fails:
+    /// - Configuration `TrustedPeers` contains entries with duplicate public keys
```

Comment:
are these two different errors or a single one?

Reply:
It's more like clarification. One of possible reasons why deserialization might fail.

Reply:
```suggestion
    /// Fails if Configuration deserialization fails (e.g. if `TrustedPeers` contains entries with duplicate public keys)
```

would this work?

Reply:
yer, sure